### PR TITLE
Improved Android Roadmap.

### PR DESCRIPTION
**Pull request by offensive-vk**
## Description

- Add Missing Content
- Add Missing Links and Resources
- Replace Old URLs
- Trim Content
- Fix Typos
- Fix Formatting and EOL
- and more !

*Thankyou.*

### CODE REVIEW

**Comment by offensive-vk** [comment]
@dansholds , how many files can i modify in this PR? (Current 15)
I wanna keep it less time consuming for you this time. 😉

**Comment by dansholds** [comment]
> @dansholds , how many files can i modify in this PR? (Current 15) I wanna keep it less time consuming for you this time. 😉

As many as you want, your last PR was clean 🤙🏻

**Comment by offensive-vk** [comment]
Thankyou @dansholds , but I will be reviewing and going through each file in `Android Roadmap` alphabetically. So its gonna be a lot of modified files (i.e 90+). I will be :

- Adding Links
- Adding / Updating Content
- Adding Anchors
- Fixing Formatting

So will that be fine for ya?

**Comment by dansholds** [comment]
> Thankyou @dansholds , but I will be reviewing and going through each file in `Android Roadmap` alphabetically. So its gonna be a lot of modified files (i.e 90+). I will be :
> 
>     * Adding Links
> 
>     * Adding / Updating Content
> 
>     * Adding Anchors
> 
>     * Fixing Formatting
> 
> 
> So will that be fine for ya?

Go for it.

**Comment by offensive-vk** [comment]
@dansholds bro, have a look 🙌

File: src/data/roadmaps/android/content/animations@Xn1VQ-xOT67ZfJJTM4r1p.md [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (dansholds) [inline src/data/roadmaps/android/content/animations@Xn1VQ-xOT67ZfJJTM4r1p.md:1]: why?

File: src/data/roadmaps/android/content/apollo-android@ww0fTbdXwVr-QIOClU7ng.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
...
\ No newline at end of file
+**Apollo Android** is a set of tools for using GraphQL with Android, made by the Apollo community developers. It's fully written in Kotlin and it was designed to seamlessly integrate with any Android app, making fetching data across network and handling data in the client-side a breeze. Apollo Android runs your queries and mutations and returns results as generated Kotlin types. It also normalizes your data and caches your results for further speed enhancements. It operates both on Android and Kotlin/JVM backend environment. It's also coroutines-first making handling concurrency easy and effective. To use Apollo Android, you'll set up the plugin, point it at your GraphQL schema, and write GraphQL queries.
+
+Visit the following resources to learn more:
+
+- [@official@Apollo Android](https://www.apollographql.com/docs/kotlin/v2/)
+- [@official@Apollo Docs](https://www.apollographql.com/docs/kotlin/)
+- [@opensource@GraphQL for Android](https://github.com/apollographql/apollo-kotlin)
```
Comment (dansholds) [inline src/data/roadmaps/android/content/apollo-android@ww0fTbdXwVr-QIOClU7ng.md:9]:
```suggestion
- [@opensource@apollographql/apollo-kotlin - GraphQL for Android](https://github.com/apollographql/apollo-kotlin)
```

File: src/data/roadmaps/android/content/animations@Xn1VQ-xOT67ZfJJTM4r1p.md (hunk 1 above)
Comment (offensive-vk) [reply src/data/roadmaps/android/content/animations@Xn1VQ-xOT67ZfJJTM4r1p.md:1]: +1 +1 ???
I don't understand

**Comment by offensive-vk** [comment]
I'll fix them real soon.

File: src/data/roadmaps/android/content/basics-of-kotlin@jl1FsQ5-WGKeFyaILNt_p.md [hunk 3]
```diff
@@ -1,3 +1,13 @@
...
\ No newline at end of file
+Kotlin is a concise, multi-platform, and fun language developed by JetBrains. Learn how to use Kotlin for Android, server-side, and cross-platform development, and explore its features, news, and community
+
+Kotlin is a statically-typed programming language that runs on the Java Virtual Machine (JVM) and can be used to develop all types of Android apps. It is Google's preferred language for Android app development.
+
+Kotlin's syntax is more concise than Java, which means less code to write and read, and fewer opportunities for errors. It provides several high-level features like lambdas, coroutines and higher order functions that help making the code more clean and understandable. Key basics of Kotlin include control flow statements (if, when, for, while), variables (mutable and non-mutable), null safety, classes and objects, inheritance, interfaces, and exception handling.
```
Comment (dansholds) [inline src/data/roadmaps/android/content/basics-of-kotlin@jl1FsQ5-WGKeFyaILNt_p.md:7]: one paragraph rather than three please

Comment (offensive-vk) [reply src/data/roadmaps/android/content/basics-of-kotlin@jl1FsQ5-WGKeFyaILNt_p.md:7]: got it m8

File: src/data/roadmaps/android/content/buttons@WhfzFOUpm0DFEj7Oeq21R.md [hunk 4]
```diff
@@ -2,6 +2,16 @@
...
 
+Buttons are UI widgets that allow users to interact with an application by tapping on them. A button typically consists of text, an icon, or a combination of both, and communicates what action will occur when the user taps it.
+
+## Types of Buttons
+
+- Text Button: A button with text only.
+- Icon Button: A button with an icon only.
+- Image Button: A button with an image only.
+- Button with Text and Icon: A button with both text and an icon.
+
```
Comment (dansholds) [inline src/data/roadmaps/android/content/buttons@WhfzFOUpm0DFEj7Oeq21R.md:13]: This isn't in line with our styling, lets let the links do the talking for this

File: src/data/roadmaps/android/content/chucker@7RKN1FNtRE_BE6QeAQrKb.md [hunk 5]
```diff
@@ -1,3 +1,8 @@
...
\ No newline at end of file
+`Chucker` is an open-source debugging library created for Android applications. It has been designed to be easy to use and convenient for developers. This library intercepts and records all HTTP requests and responses inside your application, which helps to visualize and share this information in an understandable and easy-to-read format. Using Chucker's distinct features, you can inspect all the HTTP and HTTPS traffic going in and out of your app directly. In addition, it provides other nifty features such as a user-friendly interface to view the server's raw response. It's like having a built-in network inspector in your debugging tool, enabling you to solve network-related issues more efficiently.
+
+Visit the following resources to learn more:
+
+- [@opensource@Chucker](https://github.com/ChuckerTeam/chucker)
```
Comment (dansholds) [inline src/data/roadmaps/android/content/chucker@7RKN1FNtRE_BE6QeAQrKb.md:7]:
```suggestion
- [@opensource@ChuckerTeam/chucker - Chucker On GitHub](https://github.com/ChuckerTeam/chucker)
```

File: src/data/roadmaps/android/content/buttons@WhfzFOUpm0DFEj7Oeq21R.md (hunk 4 above)
Comment (offensive-vk) [reply src/data/roadmaps/android/content/buttons@WhfzFOUpm0DFEj7Oeq21R.md:13]: aight

**Comment by offensive-vk** [comment]
Finallyyy, it's your time to shine @dansholds 😉

File: src/data/roadmaps/android/content/rxkotlin@4h37WBpYxRRyw9oH8ge7o.md [hunk 6]
```diff
@@ -1,3 +1,7 @@
...
\ No newline at end of file
+`RxKotlin` is a lightweight language extension to Java for Android development, enabling Android apps to be built using Kotlin with Reactivex. It brings the power of reactive programming paradigm to Kotlin, extending its capabilities for processing asynchronous streams of data. It allows you to express static (e.g., already known) or dynamic (e.g., future unknown) data streams, and perform various operations on them easily. Key concepts of `RxKotlin` include Observables, Observers and Schedulers. Observables represent the data streams, Observers interact with the data stream, and Schedulers determine on which thread operations are performed. RxKotlin helps manage background tasks, handle asynchronous data streams, and implement complex UIs, among others.
+
+Visit the following resources to learn more:
+
+- [@opensource@RxKotlin](https://github.com/ReactiveX/RxKotlin)
```
Comment (dansholds) [inline src/data/roadmaps/android/content/rxkotlin@4h37WBpYxRRyw9oH8ge7o.md:7]:
```suggestion
- [@opensource@ReactiveX/RxKotlin - RxKotlin on GitHub](https://github.com/ReactiveX/RxKotlin)
```

**Review by dansholds — CHANGES_REQUESTED** [review]
I've left some comments to address

**Comment by kamranahmedse** [comment]
Thank you @offensive-vk

**Pull request merged**